### PR TITLE
Add `NavigationGeocodedPlacemarkTests` tests.

### PR DESCRIPTION
**Pull request by MaximAlien**
### Description

Closing #3240.

### CODE REVIEW

File: MapboxNavigation.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -179,14 +179,15 @@
...
 		8A446645260A7B24008BA55E /* BoundingBox.swift in Sources */ = {isa = PBXBuildFile; fileRef = 8A446644260A7B24008BA55E /* BoundingBox.swift */; };
+		8A6C468226C2FA2E00452EBE /* NavigationGeocodedPlacemarkTests.swift in Sources */ = {isa = PBXBuildFile; fileRef = 8A6C468126C2FA2E00452EBE /* NavigationGeocodedPlacemarkTests.swift */; };
 		8A8C3D98260175D20071D274 /* CLLocationDirection+Camera.swift in Sources */ = {isa = PBXBuildFile; fileRef = 8A8C3D97260175D20071D274 /* CLLocationDirection+Camera.swift */; };
 		8AA849E924E722410008EE59 /* WaypointStyle.swift in Sources */ = {isa = PBXBuildFile; fileRef = 8AA849E824E722410008EE59 /* WaypointStyle.swift */; };
 		8AAE94A126A60ADE00AA1127 /* CarPlayMapViewControllerDelegate.swift in Sources */ = {isa = PBXBuildFile; fileRef = 8AAE94A026A60ADE00AA1127 /* CarPlayMapViewControllerDelegate.swift */; };
-		8AB316C926BCA56D00C3AC76 /* UIViewController.swift in Sources */ = {isa = PBXBuildFile; fileRef = 8AB316C826BCA56D00C3AC76 /* UIViewController.swift */; };
-		8AB316CB26BCA72300C3AC76 /* CGSize.swift in Sources */ = {isa = PBXBuildFile; fileRef = 8AB316CA26BCA72300C3AC76 /* CGSize.swift */; };
```
Comment (S2Ler) [inline MapboxNavigation.xcodeproj/project.pbxproj:186]: Looks like this change shouldn't be in this commit and can be just reverted, unless it appears again.

File: Tests/MapboxNavigationTests/NavigationGeocodedPlacemarkTests.swift [hunk 2]
```diff
@@ -0,0 +1,85 @@
+import XCTest
+import CarPlay
+import MapboxNavigation
+@testable import TestHelper
```
Comment (S2Ler) [inline Tests/MapboxNavigationTests/NavigationGeocodedPlacemarkTests.swift:4]: `TestHelper` is supposed to be used in tests only so if any API is  internal but needed in tests we can just make this API public.

File: Tests/MapboxNavigationTests/NavigationGeocodedPlacemarkTests.swift [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+                                                                      routableLocations: nil)
+        
+        let listItem = navigationGeocodedPlacemark.listItem()
+        XCTAssertEqual(listItem.text, title, "Titles should be equal.")
+        XCTAssertEqual(listItem.detailText, subtitle, "Subtitles should be equal.")
+        XCTAssertEqual(listItem.image, nil, "Image should not be set.")
+    }
+    
+    func testNavigationGeocodedPlacemarkEncodingAndDecoding() {
+        let coordinate = CLLocationCoordinate2D(latitude: 0.0, longitude: 0.0)
```
Comment (S2Ler) [inline Tests/MapboxNavigationTests/NavigationGeocodedPlacemarkTests.swift:25]: I think makes sense to change lat/lon to non zero values. 0 is a default value for primitive double values.

**Review by S2Ler — APPROVED** [review]

File: Tests/MapboxNavigationTests/NavigationGeocodedPlacemarkTests.swift (hunk 2 above)
Comment (MaximAlien) [reply Tests/MapboxNavigationTests/NavigationGeocodedPlacemarkTests.swift]: Yep, removed.

File: Tests/MapboxNavigationTests/NavigationGeocodedPlacemarkTests.swift (hunk 3 above)
Comment (MaximAlien) [reply Tests/MapboxNavigationTests/NavigationGeocodedPlacemarkTests.swift:25]: Don't think it's too critical at this point. Will leave it as it is.

**Pull request merged**